ReactNativeClient/lib/onedrive-api.js

```diff
@@ -317,6 +319,27 @@ class OneDriveApi {
 		throw new Error(`Could not execute request after multiple attempts: ${method} ${url}`);
 	}
 
+	setAcctProps(acctProps) {
+		this.acctProps_ = acctProps;
+	}
+
+	async execAcctPropsRequest() {
+		const response = await shim.fetch('https://graph.microsoft.com/v1.0/me/drive', {
+			method: 'GET',
+			headers: {
+				'Authorization': this.token(),
+			},
+		});
+		const data = await response.json();
+		const acctProps = { accountType: data.driveType, driveId: data.id };
+
+		if (!response.ok) {
+			const text = await response.text();
+			throw new Error(`Could not retrieve account details (drive ID, Account type): ${response.status}: ${response.statusText}: ${text}`);
+		}
```

Comment:
It's good you handle the error with clear error message. How is it display in the UI? Does it go to the log or shows in the sidebar? (ideally it should show in the sidebar).

Also I think you should check for response.ok before calling `response.json()`